query-engine/schema-builder/src/input_types/objects/update_many_objects.rs

```diff
@@ -44,16 +45,11 @@ pub(crate) fn unchecked_update_many_input_type(
     model: &ModelRef,
     parent_field: Option<&RelationFieldRef>,
 ) -> InputObjectTypeWeakRef {
-    let name = match parent_field.map(|pf| pf.related_field()) {
-        Some(ref f) => format!(
-            "{}UncheckedUpdateManyWithout{}Input",
-            model.name(),
-            capitalize(f.related_field().name())
-        ),
-        _ => format!("{}UncheckedUpdateManyInput", model.name()),
-    };
-
-    let ident = Identifier::new_prisma(name);
+    let ident = Identifier::new_prisma(IdentifierType::UncheckedUpdateManyInput(
+        model.clone(),
+        parent_field.map(|pf| pf.related_field()),
+    ));
```

Comment:
This is where the bug fix happens. We used to call `parent_field.related_field().related_field()`. That was likely a mistake.

Reply:
To elaborate further, consider the following datamodel:

```prisma
model Invoice {
    InvoiceId Int @id @default(autoincrement())

    invoice_items InvoiceItem[]
}

model InvoiceItem {
    InvoiceLineId Int @id @default(autoincrement())

    InvoiceItemInvoiceId Int @map("InvoiceId")
    invoices  Invoice @relation(fields: [InvoiceItemInvoiceId], references: [InvoiceId])

    TrackId Int
    tracks  Track @relation(fields: [TrackId], references: [TrackId])
}

model Track {
    TrackId Int    @id @default(autoincrement())
    Name    String

    invoice_items InvoiceItem[]
}
```

Notice how `Invoice.invoice_items` and `Track.invoice_items` are named the same.

The following datamodel used to produce the following input type:

```ts
prisma.Invoice.update({
  data: {
    invoice_items: <InvoiceItem>UncheckedUpdateManyWithout<Invoice_items>Input
  }
})
```

Problem is, `Track.invoice_items` would produce exactly the same type identifier.

```ts
prisma.Track.update({
  data: {
    invoice_items: <InvoiceItem>UncheckedUpdateManyWithout<Invoice_items>Input
  }
})
```

By using the related field to render the identifier name, we now generate the following input types:
- `<InvoiceItem>UncheckedUpdateManyWithout<Invoices>Input`
- `<InvoiceItem>UncheckedUpdateManyWithout<Tracks>Input`

Which lead to a new type being that did not exist before:

```json
{
  "name": "InvoiceItemUncheckedUpdateManyWithoutTracksInput",
  "constraints": {
    "maxNumFields": null,
    "minNumFields": null
  },
  "fields": [
    {
      "name": "InvoiceLineId",
      "isRequired": false,
      "isNullable": false,
      "inputTypes": [
        {
          "type": "Int",
          "location": "scalar",
          "isList": false
        },
        {
          "type": "IntFieldUpdateOperationsInput",
          "namespace": "prisma",
          "location": "inputObjectTypes",
          "isList": false
        }
      ]
    },
    {
      "name": "InvoiceItemInvoiceId",
      "isRequired": false,
      "isNullable": false,
      "inputTypes": [
        {
          "type": "Int",
          "location": "scalar",
          "isList": false
        },
        {
          "type": "IntFieldUpdateOperationsInput",
          "namespace": "prisma",
          "location": "inputObjectTypes",
          "isList": false
        }
      ]
    }
  ]
}
```

Reply:
Resolved: we are deferring the fix of this bug to Prisma 5 because it is breaking.